### DISCUSSION TITLE
How to define an interval validate callbacks in lightning

### DISCUSSION THREAD

**Discussion by rentainhe** [Lightning-AI/pytorch-lightning#9534] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I want to train my model in 20000 steps and run evaluation on each 1000 steps
I try to define the callback like this:
```
class IntervalStepValidate(Callback):
    def __init__(self, config):
        self.config = config
        self.total_steps = 20000
        self.validation_interval = 1000

    def on_batch_end(self, trainer, pl_module):
        if self.total_steps % self.validation_interval == 0:
            trainer.run_evaluation()
```
But I find out that there is no `run_evaluation()` in the latest version of Pytorch-Lightning :(
How can I update this code to get the function I want?

**Comment by rohitgr7** [comment, marked as answer]
there is a `val_check_interval` argument for it inside `Trainer`. You can set `Trainer(val_check_interval=1000)`.

**Reply by rentainhe** [reply]
Thanks very much!  It works !
And I have another question, I wonder if the `Trainer(accumulate_grad_batches=int)` can reduce the GPU memory or not ?  :)

**Reply by rohitgr7** [reply]
I don't think so. Two things here: forward and backward pass.

A forward pass isn't affected by this.
During backward pass, it will still happen and gradients will be summed until `accumulate_grad_batches`. So `optimizer.step` will eventually happen thus it won't have any effect either.

Update: if you reduce the batch_size too, then it will reduce the memory usage.

But that's just theoretical, worth a try :)

**Reply by rentainhe** [reply]
OK, I just did some quick test and find out that the `accumulate_grad_batches` can not reduce GPU memory cost :(

Cuz I don't have enough GPU memory to train my model on such a large batch-size, I have seen some repo to use `gradient_accumulation_steps` to reduce the GPU memory cost, like [ViT-pytorch](https://github.com/jeonsworld/ViT-pytorch), I have no idea why it works in that repo 😆

**Reply by rohitgr7** [reply]
While setting `accumulate_grad_batches>1`, did you reduce your batch_size? If you did then it should reduce the memory usage.

**Reply by rentainhe** [reply]
You mean, in general way, if I just want to train on a specific `batch-size 512`, I can set `batch-size=256` and `gradient-steps=2` to get the same result. 😃

**Reply by rentainhe** [reply]
Cuz I didn't see any batch split operation in [ViT-pytorch/train.py](https://github.com/jeonsworld/ViT-pytorch/blob/main/train.py), I was curious about why it works there lolllll!

**Reply by rentainhe** [reply]
Sorry! it's my fault!  He did batch split operation here:
https://github.com/jeonsworld/ViT-pytorch/blob/460a162767de1722a014ed2261463dbbc01196b6/train.py#L147
😢

**Reply by rohitgr7** [reply]
haha.. np.. it happens all the time, part of the process :)
lightning doesn't alter the batch_size internally, so yeah you have the set it from your end.

**Reply by rentainhe** [reply]
Thanks a lot ! ❤️
